compress/decompress.go

```diff
@@ -62,6 +62,13 @@ func NewDecompressor(compressedFile string) (*Decompressor, error) {
 	d := &Decompressor{
 		compressedFile: compressedFile,
 	}
+
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
this means func will return err=nil on panic and higher-level funcs will thing that operation succeed. Assign error with more info to named return parameter. Like:
```
if rec := recover(); rec != nil {
	err = fmt.Errorf("file: %s, %+v, trace: %s", compressedFile, rec, dbg.Stack())
}
```

Reply:
Reasonable. I thought I'm avoiding that effect because err declaration is lower than defer, and deferred function will not have access to that err.

Reply:
In such cases better define err right inside function declaration (named return params).